hal/halx86/legacy/bussupp.c

```diff
@@ -963,12 +965,30 @@ HalpDebugPciDumpBus(IN ULONG i,
             Mem = 0;
         if (Mem)
         {
+            ULONG PciBar = 0xFFFFFFFF;
+
+            HalpWritePCIConfig(BusHandler,
+                               PciSlot,
+                               &PciBar,
+                               FIELD_OFFSET(PCI_COMMON_HEADER, u.type0.BaseAddresses[b]),
+                               sizeof(ULONG));
+            HalpReadPCIConfig(BusHandler,
+                              PciSlot,
+                              &PciBar,
+                              FIELD_OFFSET(PCI_COMMON_HEADER, u.type0.BaseAddresses[b]),
+                              sizeof(ULONG));
+            HalpWritePCIConfig(BusHandler,
+                               PciSlot,
+                               &Mem,
+                               FIELD_OFFSET(PCI_COMMON_HEADER, u.type0.BaseAddresses[b]),
+                               sizeof(ULONG));
```

Comment:
Ah, so that means it's not possible to obtain correct BAR size without writing to it?